qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+            "Content-type": "application/json",
+            "Authorization": f"Bearer {storage_access_token}",
+        }
+
+        url = (
+            f"https://management.azure.com/subscriptions/{subscription_id}/resourceGroups/"
+            f"{resource_group_name}/providers/Microsoft.Storage/storageAccounts/{storage_account_name}"  # pylint: disable=line-too-long
+            f"/blobServices/default/containers/{container_name}?api-version=2022-09-01"
+        )
+
+        session.put(url, json={}, headers=container_headers)
+        return [job_id, container_name]
+
+    # pylint: disable=too-many-arguments
+    @staticmethod
+    def create_job_routes(
+        session: str,
+        container: str,
+        qasm: str,
+        connection_string: str,
+        location_name: str,
+        subscription_id: str,
+        resource_group_name: str,
+        workspace_name: str,
+        quantum_access_token: str,
+    ) -> list[str]:
+        """Create the routes for the job."""
+        containerName = container
+        qasm_str = qasm
+
+        blob_service_client = BlobServiceClient.from_connection_string(connection_string)
+
+        blob_client = blob_service_client.get_blob_client(containerName, "inputData")
+        blob_client.upload_blob(
+            qasm_str, content_settings=ContentSettings(content_type="application/qasm")
+        )
+
+        blob_client = blob_service_client.get_blob_client(containerName, "rawOutputData")
+        blob_client.upload_blob("")
+
+        url = (
+            f"https://{location_name}.quantum.azure.com/subscriptions/{subscription_id}"
+            f"/resourceGroups/{resource_group_name}/providers/Microsoft.Quantum/workspaces/"
+            f"{workspace_name}/storage/sasUri?api-version=2022-09-12-preview"
+        )
+
+        quantum_headers = {
+            "Content-type": "application/json",
+            "Authorization": f"Bearer {quantum_access_token}",
+        }
+
+        payload = {"containerName": containerName}
+
+        response = session.post(url, json=payload, headers=quantum_headers)
+        containerSasUri = response.json()["sasUri"]
+
+        payload["blobName"] = "inputData"
+        response = session.post(url, json=payload, headers=quantum_headers)
+        inputSasUri = response.json()["sasUri"]
+
+        payload["blobName"] = "rawOutputData"
+        response = session.post(url, json=payload, headers=quantum_headers)
+        outputSasUri = response.json()["sasUri"]
+
+        return [containerSasUri, inputSasUri, outputSasUri, quantum_headers]
+
+    # pylint: disable=too-many-arguments
+    @staticmethod
+    def create_payload(
+        container_uri: str,
+        input_uri: str,
+        output_uri: str,
+        job_id: str,
+        job_name: str,
+        input_data_format: str,
+        provider: str,
+        backend: str,
+        num_qubits: str,
+        num_shots: int,
+        total_count: int,
+        output_data_format: str,
```

Comment:
Are you sure these are all required request parameters? Or will Azure populate some of these with default values if not provided? If so, we can convert whichever are not explicitly required into optional arguments